src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,30 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new NullPointerException("Login can`t be null");
+        }
+        if (user.getPassword() == null) {
+            throw new NullPointerException("Password can`t be null");
+        }
+        if (user.getAge() == null) {
+            throw new NullPointerException("Age can`t be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("Login like this already exist");
+        }
```

Comment:
This check must be the last one, because if the password if login is not valid the user can't be found in the storage